### PR TITLE
SAK-44630 allow institution to send Sakai TAs to Turnitin as Instructors

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

**Review by ern — APPROVED** [review]
Few formatting issues but looks good otherwise

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 1]
```diff
@@ -2074,6 +2074,10 @@
 # turnitin.report_gen_speed.setting.2=1
 # turnitin.report_gen_speed.setting.3=2
 
+# Allow Sakai TAs to be sent to Turnitin as Instructors so they can view TII reports
+# DEFAULT: false
+# turnitin.enrollAssistantsAsInstructors=true
```
Comment (baholladay) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:2079]: you may want to change this to snake case to be consistent with the other permissions

File: content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java [hunk 2]
```diff
@@ -2260,23 +2269,22 @@ private String getTEM(String cid) {
 	 * @return
 	 */
 	@SuppressWarnings("unchecked")
```
Comment (baholladay) [inline content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java:2271]: `@SuppressWarnings` is not needed any longer

File: content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java [hunk 3]
```diff
@@ -2260,23 +2269,22 @@ private String getTEM(String cid) {
...
-	public Map getInstructorInfo(String siteId) {
+	public Map<String, String> getInstructorInfo(String siteId) {
 		return getInstructorInfo(siteId, false);
 	}
 
 	@SuppressWarnings("unchecked")
```
Comment (baholladay) [inline content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java:2276]: `@SuppressWarnings` is not needed any longer

File: content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java [hunk 4]
```diff
@@ -2312,14 +2320,24 @@ public Map getInstructorInfo(String siteId, boolean ignoreUseSource) {
 		return togo;
 	}
 
-	private Set<String> getActiveInstructorIds(String INST_ROLE, Site site) {
+	private Set<String> getActiveUsersForRole(final String rolePermission, final String siteId) {
```
Comment (baholladay) [inline content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java:2323]: `final` isn't needed

File: content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java [hunk 5]
```diff
@@ -2312,14 +2320,24 @@ public Map getInstructorInfo(String siteId, boolean ignoreUseSource) {
...
+		try {
+			Site site = siteService.getSite(siteId);
+			return getActiveUsersForRole(rolePermission, site);
+		} catch (IdUnusedException e) {
+			log.info("Ignoring site " + siteId + " which no longer exists.");
+		}
+
+    return new HashSet<String>();
```
Comment (baholladay) [inline content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java:2331]: spacing

File: content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java [hunk 6]
```diff
@@ -2750,39 +2768,13 @@ private User getUser(String userid) {
...
-		return users;
-	}
 	/**
 	 * The primary method of this class. Syncs the enrollment between a Sakai
 	 * Site and it's corresponding
 	 *
 	 * @param sakaiSiteID
 	 */
-	private boolean syncSiteWithTurnitin(String sakaiSiteID) {
+	private boolean syncSiteWithTurnitin(final String sakaiSiteID) {
```
Comment (baholladay) [inline content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java:2777]: if `final` needed here?

**Comment by baholladay** [comment]
LGTM... added a few minor comments

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties (hunk 1 above)
Comment (ottenhoff) [reply config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:2079]: agree

File: content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java (hunk 5 above)
Comment (ottenhoff) [reply content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java:2331]: yup tabs

File: content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java (hunk 6 above)
Comment (ottenhoff) [reply content-review/impl/turnitin/src/main/java/org/sakaiproject/contentreview/turnitin/TurnitinReviewServiceImpl.java:2777]: not really needed just helps me when doing refactoring as the IDE then lets me know if the value is being manipulated

**Comment by ottenhoff** [comment]
thanks for review. new changes pushed.

**Comment by baholladay** [comment]
LGTM

**Pull request merged**